cpp/src/cider/tests/utils/CiderArrowChecker.cpp

```diff
@@ -69,6 +69,52 @@ bool checkArrowBuffer(const struct ArrowArray* expect_array,
   return true;
 }
 
+template <typename T>
+bool checkArrowBufferDecimal(const struct ArrowArray* expect_array,
+                             const struct ArrowArray* actual_array) {
+  if (expect_array == nullptr && actual_array == nullptr) {
+    return true;
+  }
+  if (expect_array == nullptr || actual_array == nullptr) {
+    return false;
+  }
+
+  auto decimal_buffer = reinterpret_cast<const uint8_t*>(expect_array->buffers[1]);
+  auto actual_value_buffer = reinterpret_cast<const T*>(actual_array->buffers[1]);
+
+  auto expect_null_buffer = reinterpret_cast<const uint8_t*>(expect_array->buffers[0]);
+  auto actual_null_buffer = reinterpret_cast<const uint8_t*>(actual_array->buffers[0]);
+
+  if (expect_null_buffer && actual_null_buffer) {
+    for (int64_t i = 0; i < expect_array->length; i++) {
+      bool expect_valid = CiderBitUtils::isBitSetAt(expect_null_buffer, i);
+      bool actual_valid = CiderBitUtils::isBitSetAt(actual_null_buffer, i);
+      if (expect_valid != actual_valid) {
+        LOG(INFO) << "ArrowArray null bit not equal: "
+                  << "Expected: " << expect_valid << ". Actual: " << actual_valid;
+        return false;
+      }
+      if (expect_valid) {
+        T expect_value = *(reinterpret_cast<const T*>(decimal_buffer + 16 * i));
+        if (expect_value != actual_value_buffer[i]) {
+          return false;
+        }
+      }
+    }
+  } else {
+    if (!(expect_null_buffer == nullptr && actual_null_buffer == nullptr)) {
+      LOG(INFO) << "One ArrowArray null buffer is null in checkArrowBuffer.";
+    }
+    for (int64_t i = 0; i < expect_array->length; i++) {
+      T expect_value = *(reinterpret_cast<const T*>(decimal_buffer + 16 * i));
+      if (expect_value != actual_value_buffer[i]) {
+        return false;
+      }
+    }
+  }
+  return true;
+}
+
```

Comment:
keep value check in this func, encapsulate and reuse arrow buffer check part in `checkArrowBuffer`?